docgen/parser/parser.go

```diff
@@ -463,14 +463,16 @@ func generateResponseExample(messages map[string]*Message, name string) (string,
 // normalizeRequestFields is used to filter out all fields marked with an predefined indicator of a request message.
 func normalizeRequestFields(messages map[string]*Message, name string) {
 	const hidingIndicator = "docgen: hide"
-	var filtered []*Field
-	for _, f := range messages[name].Fields {
-		if strings.Contains(f.Comment.Leading, hidingIndicator) {
-			continue
+	for _, message := range messages {
```

Comment:
with this changes, does that meant that we now loop through all messages , not just the one with matching name? If yes:

* name is now unused
* I think there are duplicated efforts looping through all those messages. Could you try to move normalizeRequestFields to the top level function (ParseFile).